### ISSUE TITLE
Create an official plugin for Google Play Services

### ISSUE THREAD

**Issue by LowBudgetHomebrew** [godotengine/godot-proposals#2724]
### Describe the project you are working on
Mobile Game (Android, Google Play)
### Describe the problem or limitation you are having in your project
I can't implement Google Play Services even with https://github.com/cgisca/PGSGP which has been updated 10 month ago and seems to be abandoned
### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Something like you already did in https://docs.godotengine.org/en/3.2/tutorials/platform/android_in_app_purchases.html
I need something I can download (or find in Godot Engine as a build-in stuff) and use it.
It will help me overcome the problem of having to learn java to get PGSGP work (hours and hours of work)
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Like in PGSGP it's a singleton which will be connected as an export-plugin. Better see https://github.com/cgisca/PGSGP for details, it looks fine, but abandoned
### If this enhancement will not be used often, can it be worked around with a few lines of script?
Well, I guess you can enable or disable this plugin. If you don't need it - disable it, lol
### Is there a reason why this should be core and not an add-on in the asset library?
Well, it can be an add-on, I don't really care where I will get it, I only care of using it without having any shaman-dancing to get it work xD

**Comment by Calinou** [comment]
There is already an official plugin for Google Play Billing: https://github.com/godotengine/godot-google-play-billing

I guess a second official plugin would have to be created for Google Play Services.

@m4gr3d Any opinions on this?

**Comment by m4gr3d** [comment]
> I guess a second official plugin would have to be created for Google Play Services.

Yes that sounds like the best approach. I am actually surprised a community one doesn't exist already; there are several ones for firebase, implementing one for Google play services should be a similar effort.

@LowBudgetHomebrew As @Calinou mentioned, what you are looking for should be doable via an Android plugin.
I cannot give you a timeline however as this is outside the scope of the core developer team. The best option would be for a interested member of the community to pick up the task of providing a Google Play Services Android plugin.

**Comment by LowBudgetHomebrew** [comment]
Community already have done this, but it looks like devs have abandoned this project :(

I'll copy-paste the functionality from [PGSGP](https://github.com/cgisca/PGSGP) and highlight the things I need the most:

- **Sign-in/Sign out**
- **Achievements**
- **Leaderboards**
- Events
- Player Stats
- Player Info
- Saved Games

Today I'll try to make it work, but I'm not feeling lucky :D

**Comment by volzhs** [comment]
> I am actually surprised a community one doesn't exist already; there are several ones for firebase

I think it's because implementing firebase integration can be multi-use with one source for auth, database, etc...

**Comment by LowBudgetHomebrew** [comment]
Hi everyone! Any updates here?

**Comment by Calinou** [comment]
> Hi everyone! Any updates here?

As far as I know, nobody has started work on this yet. Considering this is a large undertaking, it may take a while to happen.

**Comment by LowBudgetHomebrew** [comment]
wellp, I just hope this feature will appear someday

**Comment by syntaxerror247** [comment]
As @m4gr3d mentioned, this is out of scope for the core team. There isn't an official plugin, but a community maintained one is now available under `godot-sdk-integrations`: https://github.com/godot-sdk-integrations/godot-play-game-services.
